### ISSUE TITLE
Ignore Epochs

### ISSUE THREAD

**Issue by jinmc** [ultralytics/yolov5#5971]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

I have noticed that my training session always stop at 100.
When I see the results of fitness, the fitness is best at epoch= 2, so it is stopping at 100, because the default patience is 100.
This is the mAP pictures, after epoch 2, the fitness decreases drastically and starts to grow gradually.
I want a feature that ignores the fitness before certain epochs and then start to count to the patience.
For example, when I set the parameter for ignore_epochs to 40, even though the fitness hasn't increased from 0, the training will be held until 140, provided the patience is 100.

### Use case

When training, especially using pretrained weights, there is a spike in the early stage of training.
I have seen this, and the only way to solve this is to manually set patience to large or not use patience.

This is when we can use ignore-epochs, so that the early epochs do not apply when counting patience.

![Screenshot 2021-12-14 at 14-43-46 Weights Biases](https://user-images.githubusercontent.com/16929516/145940212-63a59a58-1501-4491-9144-ebde99c8bf69.png)
![Screenshot 2021-12-14 at 14-43-25 Weights Biases](https://user-images.githubusercontent.com/16929516/145940213-53735d1d-06af-4643-90b7-d9a1ad9b5db6.png)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@jinmc see https://github.com/ultralytics/yolov5/pull/5303. You can update early stopping easily i.e. `python train.py --patience 300` or use `--patience 0` to disable EarlyStopping.

**Comment by jinmc** [comment]
@glenn-jocher But I want to use Earlystopping just by excluding beginning epochs, but now it doesn't seem to have that behavior.

**Comment by glenn-jocher** [comment]
@jinmc I don't think we want to add anything not in the TF implementation that I based mine on:
https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/EarlyStopping

In your example EarlyStopping seems to be working as intendend. Your best result was at epoch 2 or 3, after that your results are not improving. You probably want to train less epoch, i.e. --epochs 10, or increase your warmup epochs in your hyp file, or just evolve hyps.
